projects/com.oracle.truffle.llvm.parser.factories/src/com/oracle/truffle/llvm/parser/factories/LLVMNativeIntrinsicsProvider.java

```diff
@@ -174,17 +180,110 @@ public LLVMNativeIntrinsicFactory(boolean forceInline, boolean forceSplit) {
         protected abstract RootCallTarget generate(FunctionType type);
     }
 
+    protected static class Demangler {
+        protected final List<UnaryOperator<String>> demanglerFunctions = Arrays.asList(this::demangleRustSymbol);
+
+        protected Optional<String> demangle(String name) {
+            return demanglerFunctions.stream().map(demangler -> demangler.apply(name)).filter(n -> n != null).findFirst();
+        }
+
+        protected String demangleRustSymbol(String name) {
```

Comment:
Is this Rust specific or an LLVM IR thing?
If it's not Rust specific the name is misleading.